server/api/v1/controllers/quotesApiController.js

```diff
@@ -0,0 +1,279 @@
+            _id: quoteId,
+            text: quotePut.text,
+            originator: quotePut.originator.replace(/@/g, ""),
+            creator: quotePut.creator.replace(/@/g, ""),
+            game: quotePut.game
+        };
+
+        if (quotePut.createdAt && quotePut.createdAt != null && quotePut.createdAt !== "") {
+            newQuote.createdAt = moment(quotePut.createdAt).toISOString();
+        } else {
+            newQuote.createdAt = moment().toISOString();
+        }
+
+        const newQuoteId = await quotesManager.addQuote(newQuote);
+        quote = await quotesManager.getQuote(newQuoteId);
+
+    // If existing quote ID, overwrite
+    } else {
+        quote.text = quotePut.text;
+        quote.originator = quotePut.originator.replace(/@/g, "");
+        quote.creator = quotePut.creator.replace(/@/g, "");
+        quote.game = quotePut.game;
+    
+        if (quotePut.createdAt && quotePut.createdAt != null && quotePut.createdAt !== "") {
+            quote.createdAt = moment(quotePut.createdAt).toISOString();
+        }
+
+        try {
+            quote = await quotesManager.updateQuote(quote);
+        } catch (e) {
+            return res.status(500).send({
+               status: "error",
+               message: "Error storing quote " + quoteId + ": " + e
+            });
+        }
+    }
+
+    if (quote == null) {
+        return res.status(500).send({
+            status: "error",
+            message: "Error storing quote " + quoteId
+         });
+    }
+
+    return res.status(201).send(formatQuote(quote));
+};
+
+exports.patchQuote = async function(req, res) {
+    let { quoteId } = req.params;
+    const quotePatch = req.body;
+
+    if (quoteId == null) {
+        return res.status(400).send({
+            status: "error",
+            message: "No quoteId provided"
+        });
+    }
+
+    quoteId = parseInt(quoteId);
+    if (isNaN(quoteId)) {
+        return res.status(400).send({
+            status: "error",
+            message: "Invalid quoteId provided"
+        });
+    }
+
+    let quote = await quotesManager.getQuote(quoteId);
+
+    if (quote == null) {
+        return res.status(404).send({
+            status: "error",
+            message: "Quote " + quoteId + " not found"
+        });
+    }
+
+    if (quotePatch.text != null && quotePatch.text !== "") {
+        quote.text = quotePatch.text;
+    }
+
+    if (quotePatch.originator != null && quotePatch.originator !== "") {
+        quote.originator = quotePatch.originator.replace(/@/g, "");
+    }
+    
+    if (quotePatch.creator != null && quotePatch.creator !== "") {
+        quote.creator = quotePatch.creator.replace(/@/g, "");
+    }
+
+    if (quotePatch.game != null && quotePatch.game !== "") {
+        quote.game = quotePatch.game;
+    }
+
+    if (quotePatch.createdAt && quotePatch.createdAt != null && quotePatch.createdAt !== "") {
+        quote.createdAt = moment(quotePatch.createdAt).toISOString();
+    }
```

Comment:
We can use optional chaining to simply these null and empty checks. This feature wasn't always available for Firebot until more recently so a lot of our code doesn't use it yet but definitely worth doing so going forward.

EX:
```js
if (quotePatch.text?.length > 0) {
        quote.text = quotePatch.text;
}
```